### PR TITLE
Add category_field to AD command in PPL

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
Add `category_field` to `AD` command. PPL will group inputs by category, and each group will be sent to AD for predictions separately. This is a temporary solution before ml-commons supports 2 dimensional data.

Enabled doctest for AD (which was disabled by #575)
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java [hunk 1]
```diff
@@ -63,22 +66,37 @@ public class ADOperator extends MLCommonsOperatorActions {
   @Override
   public void open() {
     super.open();
-    DataFrame inputDataFrame = generateInputDataset(input);
+    String categoryField = arguments.containsKey(CATEGORY_FIELD)
+        ? (String) arguments.get(CATEGORY_FIELD).getValue() : null;
```
Comment (penghuo) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java:70]: avoid using null value?

**Comment by penghuo** [comment]
> ### Description
> Add `category_field` to `AD` command. PPL will group inputs by category, and each group will be sent to AD for predictions separately. This is a temporary solution before ml-commons supports 2 dimensional data.
> 
> Enabled doctest for AD (which was disabled by #575)
> 
> ### Issues Resolved
> [List any issues this PR will resolve]
> 
> ### Check List
> * [ ]  New functionality includes testing.
>   
>   * [ ]  All tests pass, including unit test, integration test and doctest
> * [ ]  New functionality has been documented.
>   
>   * [ ]  New functionality has javadoc added
>   * [ ]  New functionality has user manual doc added
> * [x]  Commits are signed per the DCO using --signoff
> 
> By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Agree, this is the temporary solution, please add the issue to track AD change in future release.

**Review by penghuo — APPROVED** [review]

File: opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java (hunk 1 above)
Comment (joshuali925) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java:70]: what do you suggest? i thought about optional but code style said to not use `Optional` as function parameters, and other arguments defaults to null https://github.com/opensearch-project/sql/blob/50d8d2da51731f0222ccaff7366ee64c5e4170b8/opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java#L128-L131

**Comment by joshuali925** [comment]
> Agree, this is the temporary solution, please add the issue to track AD change in future release.
added https://github.com/opensearch-project/ml-commons/issues/502

**Review by anirudha — APPROVED** [review]

**Pull request merged**